### PR TITLE
refactor: Upgraded to ES6

### PR DESCRIPTION
**Pull request by GentooXativa**
* Upgraded to ES6
* Removed async/lodash
* Increased node-redis version

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,42 +1,62 @@
...
+      throw new Error('config is required');
     }
 
-    function stop(cb) {
-        if (!client) return cb()
-        logger.info(format('Disconnecting from %s', config.url || config.host || '127.0.0.1'))
-        client.quit(cb)
+    if (!dependencies.logger) {
+      logger = console;
+      logger.info = console.log;
```
Comment (Betisman) [inline index.js]: Is this needed? `console.info` could work too, couldn't it?

File: index.js [hunk 2]
```diff
@@ -1,42 +1,62 @@
...
+    } else {
+      logger = dependencies.logger;
```
Comment (Betisman) [inline index.js]:
```suggestion
  ({ logger } = dependencies);
    if (!logger) {
      logger = console;
      // logger.info = console.log;
```

File: index.js [hunk 3]
```diff
@@ -1,42 +1,62 @@
...
-        start: async.seq(init, validate, start),
-        stop: stop
+    logger.info(`Connecting to ${config.url || config.host || '127.0.0.1'}`);
+    client = redis.createClient(config);
+
+    if (config.no_ready_check) {
+      connectToRedis();
+    } else {
+      await connectToRedis();
     }
```
Comment (Betisman) [inline index.js:48]: I guess this is equivalent to what we had before

File: package.json [hunk 4]
```diff
@@ -3,6 +3,11 @@
   "version": "1.0.2",
```
Comment (Betisman) [inline package.json]: We would need to bump the version, or is @UlisesGascon who will do it?

File: package.json [hunk 5]
```diff
@@ -26,4 +29,4 @@
     "url": "https://github.com/guidesmiths/systemic-redis/issues"
   },
   "homepage": "https://github.com/guidesmiths/systemic-redis#readme"
-}
+}
```
Comment (Betisman) [inline package.json:32]:
```suggestion
}

```

File: index.js (hunk 3 above)
Comment (GentooXativa) [reply index.js:48]: Same behaviour as before

File: package.json (hunk 4 above)
Comment (UlisesGascon) [reply package.json]: In this case.. I will suggest to be included in the PR @Betisman @gentooXativa

File: package.json [hunk 6]
```diff
@@ -3,6 +3,11 @@
   "version": "1.0.2",
   "description": "A systemic redis component",
   "main": "index.js",
+  "engineStrict": true,
+  "engines": {
+    "node": ">=10.0.0",
```
Comment (UlisesGascon) [inline package.json:8]: Thanks for this addition @GentooXativa !! ❤️

**Review by UlisesGascon — APPROVED** [review]
LGTM! Seems like a Major one

**Review by Betisman — APPROVED** [review]
It's a major ;)

**Pull request merged**